### PR TITLE
Add a TRAEFIK_HTTPS_ENABLE option "redirect"

### PR DESCRIPTION
**Pull request by firehist**
This PR add a `redirect` option to `TRAEFIK_HTTPS_ENABLE` for http redirect on https.

FYI: I first followed the traefik documentation about [HTTP redirect on HTTPS](https://docs.traefik.io/user-guide/examples/#http-redirect-on-https) but the result was:
- visit `http://mydomain.com`
- redirect to `https://mydomain.com:8443`

This PR uses the [regex/replacement syntax](https://docs.traefik.io/toml/#entrypoints-definition) for redirect.

### CODE REVIEW

File: root/opt/traefik/bin/traefik.toml.sh [hunk 1]
```diff
@@ -24,7 +29,13 @@ if [ $RC -eq 0 ]; then
 fi
 
 if [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xtrue" ]; then
-    TRAEFIK_ENTRYPOINTS_OPTS=${TRAEFIK_ENTRYPOINTS_HTTP}${TRAEFIK_ENTRYPOINTS_HTTPS}
+    TRAEFIK_ENTRYPOINTS_OPTS="${TRAEFIK_ENTRYPOINTS_HTTP}
+    ${TRAEFIK_ENTRYPOINTS_HTTPS}"
+    TRAEFIK_ENTRYPOINTS='"http", "https"'
+elif [ "X${TRAEFIK_HTTPS_ENABLE}" == "Xredirect" ]; then
```
Comment (firehist) [inline root/opt/traefik/bin/traefik.toml.sh]: BTW about `"X${VAR}"`.
The 'X' is used if you don't `"` the variable and the variable happens to be Null or undefined.

**Comment by digitalLumberjack** [comment]
Could be nice to have the HSTS header added too https://fr.wikipedia.org/wiki/HTTP_Strict_Transport_Security

**Comment by firehist** [comment]
@digitalLumberjack Yes, it would be cool but traefik doesn't allow us to set-up middleware easily for now. The only way I found is to dispatch this header from your backend directly.

**Comment by rawmind0** [comment]
Hi @firehist ,

i've already implemented redirect option to traefik package. I doesn't set another option to TRAEFIK_HTTPS_ENABLE, instead i've implemented it with TRAEFIK_HTTPS_ENABLE=only.

When you set TRAEFIK_HTTPS_ENABLE=only, http is redirected to https.

What do you think?? Would be ok for you?? :)

Thanks so much for the PR and the feedback....

Best regards...

**Comment by firehist** [comment]
Sorry for the non-response @rawmind0 
I changed the set-up to put a nginx in front of my traefik to more easily change configuration (as I know it better than traefik).

I use traefik to keep my pool of container behind a unique DNS name.

Thanks for your work ;)

**Comment by 4F2E4A2E** [comment]
Hi there! First of all thank you very much for bringing traefik to rancher! It's huuuge!

But why is this issue closed? There is currently no possibility to redirect http to https using non-default ports (80;443). Non-default ports will often be the case with load balancers / traefik instances.

**Currently** with the TRAEFIK_HTTPS_ENABLE=only option, the traffic gets redirected to a port which is not reachable.

**For example**,
If the firewall is forwarding to the internal non-default traefik port, then setting the above-mentioned option would lead to redirecting into a non-reachable route / URL:
http://example.com into https://example.com:8443

Any help would be much appreciated.

**Pull request closed without merging**